packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js

```diff
@@ -1,106 +1,89 @@
-		map[option.value] = option.label; // eslint-disable-line no-param-reassign
-		return map;
-	}, {});
-}
-
-function DropDownOptions({
-	options, onSelectOption, filterText, createIfNoneMatch, onCreateNew, onMouseEvent,
-	selectedIndex, noAvailableMessage,
-}) {
-	const optionsToShow = options.map((item, index) => (
-		<MenuItem active={selectedIndex === index} key={index} onClick={() => onSelectOption(item)}>
-			{item.label}
-		</MenuItem>
-	));
-	const NO_AVAILABLE = (
-		<MenuItem onClick={() => onSelectOption()}>
-			{noAvailableMessage}
-		</MenuItem>
-	);
-	const NEW_OPTION = (
-		<MenuItem active onClick={onCreateNew}>
-			{filterText} (new)
-		</MenuItem>
-	);
+const mapValue2Label = enumOptions => enumOptions.reduce((map, option) => {
+	map[option.value] = option; // eslint-disable-line no-param-reassign
+	return map;
+}, {});
 
-	function renderOptions() {
-		if (optionsToShow.length > 0) {
-			return optionsToShow;
-		} else if (createIfNoneMatch && filterText.length > 0) {
-			return NEW_OPTION;
-		}
-		return NO_AVAILABLE;
+const getAllOptions = (options) => {
+	if (options && options.groupBy) {
+		return options.enumOptions.map(opt => ({ ...opt.label, value: opt.value }));
 	}
+	return options.enumOptions;
+};
 
-	return (
-		<ul
-			role="menu"
-			className={classNames(theme['dropdown-menu'], 'dropdown-menu')}
-			onMouseEnter={() => onMouseEvent(ENTER)}
-			onMouseLeave={() => onMouseEvent(LEAVE)}
-		>
-			{renderOptions()}
-		</ul>
+const filterOptions = props => getAllOptions(props.options).filter(
+			option => props.value.indexOf(option.value) < 0
 	);
-}
 
 class MultiSelectTagWidget extends React.Component {
+
 	constructor(props) {
 		super(props);
 		this.state = {
-			showDropDownOptions: false,
-			selectedIndex: 0,
 			filterText: '',
+			withCategory: typeof props.options.groupBy !== 'undefined',
+		};
+		this.theme = {
+			container: theme.typeahead,
+			itemsContainer: classNames(theme['items-container'], 'items-container'),
 		};
 
-		this.isMouseHoverOnOptions = false;
+		this.onFocus = this.onFocus.bind(this);
+		this.onSelect = this.onSelect.bind(this);
+		this.onKeyDown = this.onKeyDown.bind(this);
+		this.onChange = this.onChange.bind(this);
+		this.getDropdownItems = this.getDropdownItems.bind(this);
+		this.updateSuggestions = this.updateSuggestions.bind(this);
+		this.resetSuggestions = this.resetSuggestions.bind(this);
+	}
 
-		this.toggleDropDownOptions = this.toggleDropDownOptions.bind(this);
-		this.onCreateNewTag = this.onCreateNewTag.bind(this);
-		this.onSelectTag = this.onSelectTag.bind(this);
-		this.onMouseEvent = this.onMouseEvent.bind(this);
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.value === this.props.value) {
+			return;
+		}
+		if (this.state.suggestions) {
+			this.updateSuggestions('', nextProps);
+		}
 	}
 
-	onFilterTextChange(event) {
-		this.setState({
-			filterText: event.target.value,
-		});
+	onComponentMount(component) {
```

Comment:
i think this should be renamed to something like `setComponentReference` since `onComponentMount` is too close to the existing React lifecyle method names